ios/FluentUI/Bottom Sheet/CommandingItem.swift

```diff
@@ -0,0 +1,122 @@
+        didSet {
+            if title != oldValue {
+                delegate?.commandingItem(self, didChangeTitleTo: title)
+            }
+        }
+    }
+
+    /// A `UIImage` to be displayed with the command.
+    @objc open var image: UIImage {
+        didSet {
+            if image != oldValue {
+                delegate?.commandingItem(self, didChangeImageTo: image)
+            }
+        }
+    }
+
+    /// A `UIImage` used when the command is represented as a button in selected state.
+    @objc open var selectedImage: UIImage? {
+        didSet {
+            if selectedImage != oldValue {
+                delegate?.commandingItem(self, didChangeSelectedImageTo: selectedImage)
+            }
+        }
+    }
+
+    /// Used in large content viewer if this command is represented using a view that cannot scale with Dynamic Type.
+    ///
+    /// When this is `nil`, `image` will be used instead.
+    @objc open var largeImage: UIImage? {
+        didSet {
+            if largeImage != oldValue {
+                delegate?.commandingItem(self, didChangeLargeImageTo: largeImage)
+            }
+        }
+    }
+
+    /// Indicates whether the command is currently on.
+    ///
+    /// When `commandType` is set to `.toggle`, the item toggles this automatically before calling `action`.
+    @objc open var isOn: Bool {
+        didSet {
+            if isOn != oldValue {
+                delegate?.commandingItem(self, didChangeOnTo: isOn)
+            }
+        }
+    }
+
+    /// Indicates whether the command is enabled.
+    @objc open var isEnabled: Bool {
+        didSet {
+            if isEnabled != oldValue {
+                delegate?.commandingItem(self, didChangeEnabledTo: isEnabled)
+            }
+        }
+    }
+
+    /// Determines the behavior of this command when its triggered.
+    @objc open var commandType: CommandType {
+        didSet {
+            if commandType != oldValue {
+                delegate?.commandingItem(self, didChangeCommandTypeTo: commandType)
+            }
+        }
+    }
+
+    @objc public init(title: String,
+                      image: UIImage,
+                      action: @escaping (CommandingItem) -> Void,
+                      selectedImage: UIImage? = nil,
+                      largeImage: UIImage? = nil,
+                      isSelected: Bool = false,
+                      isEnabled: Bool = true,
+                      commandType: CommandType = .simple) {
+        self.title = title
+        self.action = action
+        self.image = image
+        self.selectedImage = selectedImage
+        self.largeImage = largeImage
+        self.isOn = isSelected
+        self.isEnabled = isEnabled
+        self.commandType = commandType
+    }
+
+    @objc public enum CommandType: Int {
+        /// Calls `action` on tap without modifying the item.
+        case simple
+
+        /// Toggles `isOn` on tap before calling `action`.
+        case toggle
+    }
```

Comment:
probably could simplify by just representing this as a bool property instead of an enum?

Reply:
That's a good point. I guess technically we'd have to treat it as a breaking change, but in practice I'd imagine builds probably wouldn't break because clients wouldn't be exhausting all cases. I'd imagine only shared code would be doing that.

That said, I'd expect future iterations to still be buttons so maybe better for this to be a bool so future "types" can be composed together with this? Like I think it's more likely we come up with options to change the styling that would impact either CommandType than it is we come up with something that is mutually exclusive with simple and toggle